pkgs/os-specific/linux/tuxedo-keyboard/default.nix

```diff
@@ -0,0 +1,32 @@
+{ stdenv, fetchFromGitHub, kernel, kmod }:
+
+# TODO: look at the other kernel modules packages and see if you find improvements to do
+
+stdenv.mkDerivation rec {
+  name = "tuxedo-keyboard-${kernel.version}-${version}";
+  version = "2019-08-26";
+
+  src = fetchFromGitHub {
+    owner = "tuxedocomputers";
+    repo = "tuxedo-keyboard";
+    rev = "d65e76e84cfd8169591fc2a0a7c9219fa19da1b5";
+    sha256 = "1s48qpwybwh5pwqas2d1v2a7x4r97sm4hr9i4902r1d7h384bv17";
+  };
+
+  preConfigure = ''
+    substituteInPlace Makefile --replace '/lib/modules/' "${kernel.dev}/lib/modules/"
+    substituteInPlace Makefile --replace '$(shell uname -r)' "${kernel.modDirVersion}"
+  '';
+
+  installPhase = ''
+    mkdir -p "$out/lib/modules/${kernel.modDirVersion}"
+    mv src/tuxedo_keyboard.ko $out/lib/modules/${kernel.modDirVersion}
+  '';
+
+  meta = {
+    description = "Full color keyboard driver for tuxedo computers laptops";
+    homepage = "https://github.com/tuxedocomputers/tuxedo-keyboard/";
+    platforms = stdenv.lib.platforms.linux;
+    maintainers = [ stdenv.lib.maintainers.blanky0230 ];
```

Comment:
License is missing

Reply:
38640353a67a89294ec3c6ffdaa639591d40bdd8